### PR TITLE
[risk=low][no ticket] Upgrade some UI packages and remove unused custom-elements

### PR DESCRIPTION
**Pull request by jmthibault79**
Bump `@types/enzyme` to latest - includes #8185 
Bump `@types/lodash` to latest - includes #8191 
Bump `react-modal` to latest - includes #8192 
Bump `validate.js` to latest - includes #8180
Remove unused `@webcomponents/custom-elements` instead of upgrading (#8179)

Tested locally with no obvious problems.

Tested react-modal by interacting with a modal, which worked normally.
Tested validate.js by interacting with a form which uses validation.  this worked normally.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: ui/src/index.ts [hunk 1]
```diff
@@ -4,7 +4,6 @@ import 'app/styles/genome-extraction-datatable.css';
 import 'app/styles/sidebar.css';
 import '@clr/icons/clr-icons.min.css';
 import '@clr/icons/clr-icons.min.js';
-import '@webcomponents/custom-elements/custom-elements.min.js';
```
Comment (jmthibault79) [inline ui/src/index.ts:7]: I can't find anywhere that this is used

**Review by Peter-Lavigne — COMMENTED** [review]
LGTM.

I found [the PR that adds custom elements](https://github.com/all-of-us/workbench/pull/62). It seems like a peer dependency of clarity, which we still use for some icons. I see the following warning when installing packages: `warning " > @clr/icons@0.11.34" has unmet peer dependency "@webcomponents/custom-elements@^1.0.0".`

However, icons still seem to be working. I'm unsure whether this is a local caching issue or not. I'll look more into this when I have some time.

**Comment by jmthibault79** [comment]
I also can't find any problems with `clr-icon`s after this change.

**Review by Peter-Lavigne — APPROVED** [review]
Agreed. I can't find any issues, either.

**Pull request merged**